packages/gatsby/src/query/page-query-runner.js

```diff
@@ -197,9 +157,61 @@ const runInitialQueries = async activity => {
       ).toFixed(2)} queries/second`
     )
   })
+}
+
+const processQueries = async (queryJobs, activity) => {
+  const queue = queryQueue.makeBuild()
+  reportStats(queue, activity)
   await queryQueue.processBatch(queue, queryJobs)
 }
 
+const makeStaticQueryJob = (state, queryId) => {
+  const component = state.staticQueryComponents.get(queryId)
+  const { hash, jsonName, query, componentPath } = component
+  return {
+    id: hash,
+    hash,
+    jsonName,
+    query,
+    componentPath,
+    context: { path: jsonName },
+  }
+}
+
+const processStaticQueries = async (queryIds, { state, activity }) => {
+  state = state || store.getState()
+  await processQueries(
+    queryIds.map(id => makeStaticQueryJob(state, id)),
+    activity
+  )
+}
+
+const makePageQueryJob = (state, queryId) => {
+  const page = state.pages.get(queryId)
```

Comment:
could this be null? there is an falsy check in the old code. Unsure if a query is being marked as dirty on deletion which might make this null.

Reply:
Theoretically no. `runQueuedQueries` runs in a single code block. So there is no chance for any other code to run between when it calculates the query IDs, and when it creates the query jobs.

```js
const runQueuedQueries = () => {
  if (listenerQueue) {
    const state = store.getState()
    const { staticQueryIds, pageQueryIds } = groupQueryIds(
      calcDirtyQueryIds(state)
    )
    const queryJobs = [
      ...staticQueryIds.map(id => makeStaticQueryJob(state, id)),
      ...pageQueryIds.map(id => makePageQueryJob(state, id)),
    ]
    listenerQueue.push(queryJobs)
  }
}
```

Reply:
Ah yeah 🤦‍♂️ 👍